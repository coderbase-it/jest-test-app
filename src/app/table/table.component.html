<table class="table table-dark">
  <thead>
    <tr>
      <th scope="col" *ngFor="let header of headers">{{header}}</th>
    </tr>
  </thead>
  <tbody>
   <ng-content></ng-content>
  </tbody>
</table>
