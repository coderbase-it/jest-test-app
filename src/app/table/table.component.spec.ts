import { Component } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { TableComponent } from './table.component';


@Component({
  selector: 'app-test',
  template: `<app-table [headers]="testHeaders">
    <tr>
    <th scope="row">1</th>
    <td>Mark</td>
    <td>Otto</td>
    <td>@mdo</td>
  </tr>
  <tr>
    <th scope="row">2</th>
    <td>Mark</td>
    <td>Otto</td>
    <td>@mdo</td>
  </tr>
  </app-table>`
})
class TestComponent {
  testHeaders = ['col1', 'col2']
}


describe('TableComponent', () => {
  let component: TableComponent;
  let fixture: ComponentFixture<TableComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ TableComponent, TestComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TableComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });


  it( `To check how many headers is create based on the input array provide Also check the title of each header`, () => {
    component.headers =  ['col1', 'col2']
    fixture.detectChanges();
    const columns = fixture.nativeElement.querySelectorAll('th')
    expect(columns.length).toEqual(2)
    expect(columns[0].textContent).toContain(component.headers[0])
    expect(columns[1].textContent).toContain(component.headers[1])
    console.log(fixture.nativeElement.innerHTML)

  })

  /**
   * Snapshot Jest Testing
   */

  it('snapshot test', () => {
    component.headers =  ['col1', 'col2']
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot()
  })

  it(`to test the project content on tbody part `, ()=> {
    const fixtureTest = TestBed.createComponent(TestComponent);
    const componentTest = fixtureTest.componentInstance;
    fixtureTest.detectChanges();

    const template = fixtureTest.debugElement.query(By.css('app-table')).nativeElement
    const instanceTable = fixtureTest.debugElement.query(By.css('app-table')).componentInstance

    expect(instanceTable.headers).toStrictEqual(componentTest.testHeaders)

    const lines = fixtureTest.nativeElement.querySelectorAll('tbody tr')
    expect(lines.length).toEqual(2)





  })
})

