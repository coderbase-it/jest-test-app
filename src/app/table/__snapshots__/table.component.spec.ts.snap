// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent snapshot test 1`] = `
<app-table
  __ngContext__={[Function LRootView]}
  headers={[Function Array]}
>
  <table
    class="table table-dark"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          col1
        </th>
        <th
          scope="col"
        >
          col2
        </th>
        
      </tr>
    </thead>
    <tbody />
  </table>
</app-table>
`;
